src/main/java/com/barterbay/app/servcie/registration/UserRegistrationService.java

```diff
@@ -0,0 +1,68 @@
+package com.barterbay.app.servcie.registration;
+
+import com.barterbay.app.domain.User;
+import com.barterbay.app.domain.dto.user.UserRegistrationDTO;
+import com.barterbay.app.exception.registration.OverdueTokenException;
+import com.barterbay.app.repository.UserRepository;
+import com.barterbay.app.servcie.email.GmailService;
+import lombok.AllArgsConstructor;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+@Transactional
+public class UserRegistrationService {
+
+  private final GmailService gmailService;
```

Comment:
Unless? I think we talked about the fact that we need an interface because we will have different mails. And since we are currently working with Gmail, I use the implementation GmailService. 
Am I wrong?